elements/pl-order-blocks/dag_checker.py

```diff
@@ -1,72 +1,50 @@
-
-class AnswerState:
-    def __init__(self):
-        self.valid = set()  # a block is valid if all blocks that it depends on have appeared before it
-        self.current_group = None
-        self.current_group_len = 0
-
-    def begin_group(self, group_id):
-        self.current_group = group_id
-
-    def end_group(self):
-        self.current_group = None
-        self.current_group_len = 0
-
-    def set_valid(self, stmt_id):
-        self.valid.add(stmt_id)
-        if self.current_group is not None:
-            self.current_group_len += 1
-
-    def is_valid(self, stmt_id):
-        return stmt_id in self.valid
-
-    def get_current_group_blocks(self):
-        return self.current_group_len
+from collections import Counter
+import networkx as nx
+
+
+def check_topological_sorting(order, graph):
+    seen = set()
+    for i, node in enumerate(order):
+        if node is None or not all([u in seen for (u, _) in graph.in_edges(node)]):
```

Comment:
Nit: it's more efficient to use a generator expression instead of a list comprehension here:
```suggestion
        if node is None or not all(u in seen for (u, _) in graph.in_edges(node)):
```